src/Common/test/Certificates.Test/ConfigureCertificateOptionsTest.cs

```diff
@@ -17,176 +18,287 @@ public sealed class ConfigureCertificateOptionsTest
 {
     private const string CertificateName = "test";
 
-    [Fact]
-    public void ConfigureCertificateOptions_NoPath_NoCertificate()
+    [Theory]
+    [InlineData("")]
+    [InlineData(CertificateName)]
+    public void ConfigureCertificateOptions_NoPath_NoCertificate(string certificateName)
     {
         var configureOptions = new ConfigureCertificateOptions(new ConfigurationBuilder().Build());
         var options = new CertificateOptions();
 
-        configureOptions.Configure(CertificateName, options);
+        configureOptions.Configure(certificateName, options);
 
         options.Certificate.Should().BeNull();
     }
 
-    [Fact]
-    public void ConfigureCertificateOptions_BadPath_NoCertificate()
+    [Theory]
+    [InlineData("")]
+    [InlineData(CertificateName)]
+    public void ConfigureCertificateOptions_BadPath_NoCertificate(string certificateName)
     {
-        IConfigurationRoot configurationRoot = new ConfigurationBuilder().AddInMemoryCollection(new Dictionary<string, string?>
+        IConfiguration configuration = new ConfigurationBuilder().AddInMemoryCollection(new Dictionary<string, string?>
         {
-            [$"{CertificateOptions.ConfigurationKeyPrefix}:{CertificateName}:certificateFilePath"] = "does-not-exist.crt"
+            [$"{GetConfigurationKey(certificateName, "CertificateFilePath")}"] = "does-not-exist.crt"
         }).Build();
 
-        var configureOptions = new ConfigureCertificateOptions(configurationRoot);
+        var configureOptions = new ConfigureCertificateOptions(configuration);
 
         var options = new CertificateOptions();
 
-        configureOptions.Configure(CertificateName, options);
+        configureOptions.Configure(certificateName, options);
 
         options.Certificate.Should().BeNull();
     }
 
-    [Fact]
-    public void ConfigureCertificateOptions_EmptyFile_Crashes()
+    [Theory]
+    [InlineData("")]
+    [InlineData(CertificateName)]
+    public void ConfigureCertificateOptions_EmptyFile_Crash_logged(string certificateName)
     {
-        IConfigurationRoot configurationRoot = new ConfigurationBuilder().AddInMemoryCollection(new Dictionary<string, string?>
+        CapturingLoggerProvider loggerProvider = new();
+        using var loggerFactory = new LoggerFactory([loggerProvider]);
+
+        IConfiguration configuration = new ConfigurationBuilder().AddInMemoryCollection(new Dictionary<string, string?>
```

Comment:
Lines like this are more readable when broken up, such as:
```c#
var appSettings = new Dictionary<string, string?>
{
    [$"{GetConfigurationKey(certificateName, "CertificateFilePath")}"] = "empty.crt"
};

IConfiguration configuration = new ConfigurationBuilder().AddInMemoryCollection(appSettings).Build();
```

I realize it was already like this, but I cleaned up many of them in Management. Here's an opportunity to do the same and improve the codebase.